lib/wor/paginate/formatters/ams_formatter.rb

```diff
@@ -0,0 +1,17 @@
+module Wor
+  module Paginate
+    module Formatters
+      class AmsFormatter < Base
+        def serialized_content
+          super
+          return paginated_content.as_json unless defined? serializable_resource
```

Comment:
This line confused me a little, Are you checking if the application has AMS defined, right? Would that bring a problem if the application is using multiple serializers?

Reply:
I think you can move `paginated_content.as_json` to a different formater.
```ruby
class AsJsonFormater < Base
def serialized_content
paginated_content.as_json
end
```

Reply:
Anyway serializer can be selected in the settings

Reply:
The line that buffles me is this one:

```ruby
defined? serializable_resource
```

Since you already defined the method in the same class and it calls the module, if the module is undefined it should return nil otherwise it should return the call to the module, the problem with this is that an application may have different serializers gems (I know it's not the norm but I have seen it), so this may bring some problems when calling `as_json` in an object we don't really know if it can respond to it.

If you can select the serializer from the configuration I don't see the point on having this check since you already know that they are going to implement pagination with AMS.

Reply:
If you are going to have this check, throw some custom error letting the user know that the gem was unable to format due to the AMS missing required dependencies

Reply:
@nicozar95  @mnmallea  I have this check to throw an error when AMS is not defined but I don't create now but I would like to create a generic module that I can use in all formatters